src/Makefile

```diff
@@ -129,6 +130,18 @@ endif
 # Include paths to dependencies
 FINAL_CFLAGS+= -I../deps/hiredis -I../deps/linenoise -I../deps/lua/src
 
+ifneq ($(NO_SYSTEMD),1)
+# If we're on Linux and pkg-config indicates libsystemd presence,
+# we want to link against it for sd_notify support
+ifeq ($(uname_S),Linux)
+LIBSYSTEMD_PKGCONFIG := $(shell sh -c '$(PKG_CONFIG) --silence-errors --libs libsystemd >/dev/null; echo $$?')
```

Comment:
There's no need for the `sh -c` construct here. I'd also use `&&` instead of `;`, to avoid any future switch in the Makefiles to consider shell errors fatal. Using then `echo yes`, seems more aligned with preexisting style too. And to check for the module presence you can just use `--exists`. If we want to check for a minimum version then `--atleast-version=`_VERSION_.

Also perhaps this variable should be called `HAVE_SYSTEMD`? Then `USE_SYSTEMD` could also contain `auto`, which would autodetect the presence and disable silently, or `yes` which would fail the build if it's missing, so that distributors can select which mode to use.